docs/icicle/primitives/msm.md

```diff
@@ -0,0 +1,51 @@
+# MSM
+
+MSM (Multi scalar multiplication) is a important primitive in ZK protocols.
+
+### Using MSM
+
+Our MSM implementation supports two algorithms.
+
+-   Bucket accumulation 
+-   Large triangle accumulation
+
+In most scenarios you should be using Bucket accumulation as its a more parallel friendly algorithm and will yield better performance on larger MSMs. 
+
+The exception being a large batch of smaller MSMs.
+
+```rust
+...
```

Comment:
These look like code ticks that weren't formatted correctly...maybe its just me 🤷🏻 
```suggestion
.
.
.
```